### PR TITLE
using pd.concat() instead of pd.merge()

### PR DESCRIPTION
**Pull request by AlexDo1**
Always merge Split datasets additionally.

The default parameters of [pd.concat()](https://pandas.pydata.org/docs/reference/api/pandas.concat.html) should fit our use case.

### CODE REVIEW

File: metacatalog/util/results.py [hunk 1]
```diff
@@ -383,8 +383,8 @@ def get_data(self, verbose=False, merge=False, **kwargs) -> dict:
                 elif len(unmerged) > 0:
                     data[member.checksum] = unmerged
         
-        # Composites always try to merge
-        if self.group is not None and self.group.type.name == 'Composite':
+        # Composites and Split datasets always try to merge
+        if self.group is not None and self.group.type.name in ('Composite', 'Split dataset'):
```
Comment (mmaelicke) [inline metacatalog/util/results.py:387]: I think we need to handle Composites and Split datasets separately, as composites should be merged and split datasets should be concatenated

File: metacatalog/util/results.py [hunk 2]
```diff
@@ -396,7 +396,7 @@ def get_data(self, verbose=False, merge=False, **kwargs) -> dict:
             # merge everything that is a DataFrame
             for checksum, dat in data.items():
                 if isinstance(dat, pd.DataFrame):
-                    all_data = pd.merge(all_data, dat, left_index=True, right_index=True, how='outer')
+                    all_data = pd.concat([all_data, dat])
```
Comment (mmaelicke) [inline metacatalog/util/results.py:399]: I think this only works for Split datasets as Composites were handled correctly. For a composite, the data resulting from different entries should be *merged* along a common index axis

**Review by mmaelicke — CHANGES_REQUESTED** [review]
Hey, thanks. But I think this needs some more logic to be only applied to `Split datasets` and not to `Composites`.

**Comment by AlexDo1** [comment]
The latest commit uses the 'old' `pd.merge()` function for `Composites` and `pd.concat` for `Split datasets` now.

**Composite: Eddy data**:
image

**Split dataset: LUBW gauge data**:
image

**Review by mmaelicke — COMMENTED** [review]
That looks way better now, thanks a lot!
Is there an easy way to add a unit-test for the changed behavior? If yes, that would be cool, if not, never mind and merge.

**Comment by AlexDo1** [comment]
> That looks way better now, thanks a lot!
> Is there an easy way to add a unit-test for the changed behavior? If yes, that would be cool, if not, never mind and merge.

I think you tested most of the merging behavior of `Split datasets` here:
https://github.com/VForWaTer/metacatalog/blob/3c547d4ab06178a41c347bc23afd1cf5dfa17f22/metacatalog/test/test_array_type_data.py#L246-L280

I just copied most of that test for a second test with a `Composite dataset` and tested for the shape of the merged data.
I guess we could test for more stuff here, I just focused on the merging behavior.

**Comment by AlexDo1** [comment]
From our conversation above, I understand that the tests are not perfect, but we should still be able to merge the PR since the results look good.

**Comment by mmaelicke** [comment]
It's not so easy to continue a discussion after 10 months :) But I agree, it should be fine to merge.

**Pull request merged**